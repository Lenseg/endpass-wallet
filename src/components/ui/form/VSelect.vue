<template>
  <div class="field">
    <label class="label"
           v-if="label">{{ label }}</label>
    <div class="control select">
      <select v-model="selected" :name="name"
      :data-vv-as="label || name" v-validate="validator">
        <option v-for="item in options"
                :key="item.val || item"
                :value="item.val || item">{{ item.text || item }}</option>
      </select>
    </div>
    <p class="help is-danger"
       v-if="error || errors && errors.has(name) ">{{ error || errors && errors.first(name) }}</p>
  </div>
</template>

<script>
export default {
  name: 'v-select',
  props: {
    value: {
      default: null,
    },
    validator: {
      type: String,
      default: '',
    },
    error: {
      type: String,
      default: null,
    },
    name: {
      type: String,
      default: '',
    },
    options: {
      type: Array,
      default: () => [],
    },
    label: {
      type: String,
      default: null,
    },
  },
  inject: {
    $validator: '$validator',
  },
  computed: {
    selected: {
      get() {
        return this.value;
      },
      set(newVal) {
        this.$emit('input', newVal);
      },
    },
  },
};
</script>

<style lang="scss">
</style>
