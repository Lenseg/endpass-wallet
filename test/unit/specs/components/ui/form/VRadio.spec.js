import { shallow, createLocalVue } from '@vue/test-utils';
import VeeValidate from 'vee-validate';

import VSelect from '@/components/ui/form/VRadio.vue';

const localVue = createLocalVue();

describe('VSelect', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(VSelect, {
      localVue,
      provide: () => ({
        $validator: new VeeValidate.Validator(),
      }),
    });
  });

  it('should render props', () => {
    expect(wrapper.contains('label')).toBeFalsy();
    expect(wrapper.contains('input')).toBeFalsy();

    wrapper.setProps({
      options: ['option1', 'option2'],
      value: 'option1',
      label: 'Some Label',
    });

    expect(wrapper.find('label').text()).toBe('Some Label');
    expect(wrapper.findAll('input').length).toBe(2);
  });

  it('should emit event', () => {
    wrapper.setProps({
      options: ['option1', 'option2'],
      value: 'option2',
      label: 'Some Label',
    });
    expect(wrapper.emitted().input).toBeTruthy();
  });

  describe('props', () => {
    describe('options', () => {
      describe('options are strings', () => {
        it('should correctly render options', () => {
          const options = ['option1', 'option2'];

          wrapper.setProps({ options });

          expect(wrapper.find('.v-radio').html()).toMatchSnapshot();
        });
      });

      describe('options are objects', () => {
        it('should correctly render text of options', () => {
          const options = [{ text: 'text' }, {}];

          wrapper.setProps({ options });

          expect(wrapper.find('.v-radio').html()).toMatchSnapshot();
        });

        it('should correctly render values of options', () => {
          const options = [{ val: 'val' }, {}];

          wrapper.setProps({ options });

          expect(wrapper.find('.v-radio').html()).toMatchSnapshot();
        });
      });

      describe('options have different types', () => {
        it('should correctly render options', () => {
          const options = [
            'option',
            { val: 'option value', text: 'option text' },
          ];

          wrapper.setProps({ options });

          expect(wrapper.find('.v-radio').html()).toMatchSnapshot();
        });
      });
    });
  });
});
