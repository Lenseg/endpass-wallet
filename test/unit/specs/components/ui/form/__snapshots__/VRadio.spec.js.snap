// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSelect props options options are objects should correctly render text of options 1`] = `
<div class="field is-horizontal has-addons v-radio">
    <!---->
    <div class="control">
        <label for="[object Object]" class="button is-multiline">

            <!---->
        </label>
        <input id="[object Object]" type="radio" name="" value="">
    </div>
    <div class="control">
        <label for="[object Object]" class="button is-multiline">

            <!---->
        </label>
        <input id="[object Object]" type="radio" name="" value="">
    </div>
    <!---->
</div>
`;

exports[`VSelect props options options are objects should correctly render values of options 1`] = `
<div class="field is-horizontal has-addons v-radio">
    <!---->
    <div class="control">
        <label for="val" class="button is-multiline">

            <!---->
        </label>
        <input id="val" type="radio" name="" value="val">
    </div>
    <div class="control">
        <label for="[object Object]" class="button is-multiline">

            <!---->
        </label>
        <input id="[object Object]" type="radio" name="" value="">
    </div>
    <!---->
</div>
`;

exports[`VSelect props options options are strings should correctly render options 1`] = `
<div class="field is-horizontal has-addons v-radio">
    <!---->
    <div class="control">
        <label for="option1" class="button is-multiline">
            option1
            <!---->
        </label>
        <input id="option1" type="radio" name="" value="option1">
    </div>
    <div class="control">
        <label for="option2" class="button is-multiline">
            option2
            <!---->
        </label>
        <input id="option2" type="radio" name="" value="option2">
    </div>
    <!---->
</div>
`;

exports[`VSelect props options options have different types should correctly render options 1`] = `
<div class="field is-horizontal has-addons v-radio">
    <!---->
    <div class="control">
        <label for="option" class="button is-multiline">
            option
            <!---->
        </label>
        <input id="option" type="radio" name="" value="option">
    </div>
    <div class="control">
        <label for="option value" class="button is-multiline">

            <!---->
        </label>
        <input id="option value" type="radio" name="" value="option value">
    </div>
    <!---->
</div>
`;
